components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2715,6 +2715,19 @@ where
         fail_point!("on_split", self.ctx.store_id() == 3, |_| {});
 
         let region_id = derived.get_id();
+
+        // Group in-memory pessimistic locks in the original region into new regions. The locks of
+        // new regions will be put into the corresponding new regions later. And the locks belonging
+        // to the old region will stay in the original map.
+        let region_locks = {
+            let mut pessimistic_locks = self.fsm.peer.txn_ext.pessimistic_locks.write();
+            // Update the version so the concurrent reader will fail due to EpochNotMatch
+            // instead of PessimisticLockNotFound.
+            pessimistic_locks.version = derived.get_region_epoch().get_version();
```

Comment:
L2736 sets the `pessimistic_locks` fields for the current peer, is this duplicated?

Reply:
No, it is intended. Setting this version must be protected with the `RwLock` locked. Otherwise, a concurrent reader may find some lock missing while not realizing the epoch version has changed. 

This is covered by the `test_split_pessimistic_locks_with_concurrent_prewrite` test.